modules/library/render/src/main/java/org/geotools/renderer/composite/BlendComposite.java

```diff
@@ -0,0 +1,653 @@
+                //        = Sa.Da + Dca.(1 - Sa)
+                //        = (Sa + 1 - Sa) * Da = Da
+                // otherwise if Sca == 0
+                //   Dca' = Sca.(1 - Da) + Dca.(1 - Sa)
+                //        = Dca.(1 - Sa)
+                // otherwise if Sca > 0
+                // Dca' = Sa.Da - min(Sa.Da, (Sa.Da - Dca.Sa)/Sca.Da) + Sca.(1 - Da) + Dca.(1 - Sa)
+                //
+                // Da' = Sa + Da - Sa.Da
+                // @formatter:on
+                int s1a = UBYTE_MAX_VALUE - sa;
+                int d1a = UBYTE_MAX_VALUE - da;
+                int drsa = dr * sa;
+                int dgsa = dg * sa;
+                int dbsa = db * sa;
+                int srda = sr * da;
+                int sgda = sg * da;
+                int sbda = sb * da;
+                int sada = sa * da;
+
+                result[RED] = (((srda + drsa <= sada) ? sr * d1a + dr * s1a : (sr > 0 ? sa
+                        * (srda + drsa - sada) / sr + sr * d1a + dr * s1a + UBYTE_MAX_VALUE : 0)) >> SHIFT8);
+                result[GREEN] = (((sgda + dgsa <= sada) ? sg * d1a + dg * s1a : (sg > 0 ? sa
+                        * (sgda + dgsa - sada) / sg + sg * d1a + dg * s1a + UBYTE_MAX_VALUE : 0)) >> SHIFT8);
+                result[BLUE] = (((sbda + dbsa <= sada) ? sb * d1a + db * s1a : (sb > 0 ? sa
+                        * (sbda + dbsa - sada) / sb + sb * d1a + db * s1a + UBYTE_MAX_VALUE : 0)) >> SHIFT8);
+                result[ALPHA] = (sa + da - ((sada + UBYTE_MAX_VALUE) >> SHIFT8));
+            }
+
+        },
+        HardLight("hard-light") {
+
+            @Override
+            public void perform(int sr, int sg, int sb, int sa, int dr, int dg, int db, int da,
+                    int[] result) {
+                // @formatter:off
+                // if 2.Sca < Sa
+                //   Dca' = 2.Sca.Dca + Sca.(1 - Da) + Dca.(1 - Sa)
+                // otherwise
+                //   Dca' = Sa.Da - 2.(Da - Dca).(Sa - Sca) + Sca.(1 - Da) + Dca.(1 - Sa)
+                //
+                // Da' = Sa + Da - Sa.Da
+                // @formatter:on
+                int s1a = UBYTE_MAX_VALUE - sa;
+                int d1a = UBYTE_MAX_VALUE - da;
+                int sada = sa * da;
+
+                result[RED] = (((2 * sr < sa) ? 2 * sr * dr + sr * d1a + dr * s1a : sada - 2
+                        * (da - dr) * (sa - sr) + sr * d1a + dr * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[GREEN] = (((2 * sg < sa) ? 2 * sg * dg + sg * d1a + dg * s1a : sada - 2
+                        * (da - dg) * (sa - sg) + sg * d1a + dg * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[BLUE] = (((2 * sb < sa) ? 2 * sb * db + sb * d1a + db * s1a : sada - 2
+                        * (da - db) * (sa - sb) + sb * d1a + db * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[ALPHA] = (sa + da - ((sa * da + UBYTE_MAX_VALUE) >> SHIFT8));
+            }
+
+        },
+        SoftLight("soft-light") {
```

Comment:
Is is really necessary to repeat it for each and every example? I think I understood what you wanted from the first one ;-)